### PR TITLE
✨Autoscaling: have a buffer of machines always ready (episode I)  (⚠️ devops)

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- some refactoring
- some bugfixes (issues with timezones, never use utcnow() anymore but now(timezone.utc), getting EC2 instances AND vs OR operation, sending of progress when upscaling)
- adds Name tag on EC2 instances such that lucky people with UI access can see machine names ;) @Surfict 
-  (⚠️ devops) introduces new ENV ```EC2_INSTANCES_MACHINES_BUFFER```, defaults to 0 when **undefined** Please currently define to 0

This PR teaches the autoscaling to keep a buffer of machines when upscaling and downscaling **BUT** not usable until episode II as this will indeed create more machines than necessary in case of load and keep always that number of buffer machines when scaling down but not ensure that EC2_INSTANCES_MACHINES_BUFFER is always present.

### CODE REVIEW

File: services/autoscaling/src/simcore_service_autoscaling/api/health.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 @router.get("/", include_in_schema=True, response_class=PlainTextResponse)
 async def health_check():
     # NOTE: sync url in docker/healthcheck.py with this entrypoint!
-    return f"{__name__}.health_check@{datetime.utcnow().isoformat()}"
+    return f"{__name__}.health_check@{datetime.datetime.now(datetime.timezone.utc).isoformat()}"
```
Comment (GitHK) [inline services/autoscaling/src/simcore_service_autoscaling/api/health.py:24]: I don't get it what's the difference here?

File: services/autoscaling/tests/unit/test_dynamic_scaling_core.py [hunk 2]
```diff
@@ -164,17 +161,56 @@ async def test_cluster_scaling_from_labelled_services_with_no_services_does_noth
...
     mock_start_aws_instance.assert_not_called()
-    mock_terminate_instance.assert_not_called()
+    mock_terminate_instances.assert_not_called()
     _assert_rabbit_autoscaling_message_sent(
         mock_rabbitmq_post_message, app_settings, initialized_app
     )
 
 
+# async def test_cluster_scaling_from_labelled_services_with_no_services_and_machine_buffer_starts_expected_machines(
+#     minimal_configuration: None,
```
Comment (GitHK) [inline services/autoscaling/tests/unit/test_dynamic_scaling_core.py:176]: Do you still need this?

**Review by GitHK — COMMENTED** [review]
👍

File: services/autoscaling/tests/unit/test_dynamic_scaling_core.py (hunk 2 above)
Comment (sanderegg) [reply services/autoscaling/tests/unit/test_dynamic_scaling_core.py:176]: yes, for episode II

File: services/autoscaling/src/simcore_service_autoscaling/api/health.py (hunk 1 above)
Comment (sanderegg) [reply services/autoscaling/src/simcore_service_autoscaling/api/health.py:24]: read this [https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow)

image

File: services/autoscaling/tests/unit/test_dynamic_scaling_core.py (hunk 2 above)
Comment (GitHK) [reply services/autoscaling/tests/unit/test_dynamic_scaling_core.py:176]: k, maybe put a note on it

Comment (sanderegg) [reply services/autoscaling/tests/unit/test_dynamic_scaling_core.py:176]: it comes right after this one.

File: services/autoscaling/src/simcore_service_autoscaling/api/health.py (hunk 1 above)
Comment (pcrespov) [reply services/autoscaling/src/simcore_service_autoscaling/api/health.py:24]: interesting. We should propagate this to the entire repo. We us the "utcnow" everywhere! Perhaps create an issue to not forgetting refactoring it

**Review by pcrespov — APPROVED** [review]

**Review by GitHK — APPROVED** [review]
Nice!

File: services/autoscaling/src/simcore_service_autoscaling/api/health.py (hunk 1 above)
Comment (GitHK) [reply services/autoscaling/src/simcore_service_autoscaling/api/health.py:24]: Agree, I'd like to have a repo wide "timezone aware etc now"

**Comment by mrnicegyu11** [comment]
Sorry I absolutely do not have time to even quickly read this at this point. just letting you know.

**Pull request merged**